controllers/etcd_controller.go

```diff
@@ -840,31 +612,48 @@ func (r *EtcdReconciler) reconcileEtcd(ctx context.Context, logger logr.Logger,
 		return nil, nil, fmt.Errorf("Spec.Replicas should not be even number: %d", etcd.Spec.Replicas)
 	}
 
-	val := componentetcd.Values{
-		ConfigMap: componentconfigmap.GenerateValues(etcd),
-		Lease:     componentlease.GenerateValues(etcd),
-		Service:   componentservice.GenerateValues(etcd),
+	etcdImage, etcdBackupImage, err := getEtcdImages(r.ImageVector, etcd)
+	if err != nil {
+		return nil, nil, err
 	}
 
-	leaseDeployer := componentlease.New(r.Client, etcd.Namespace, val.Lease)
+	if etcd.Spec.Etcd.Image == nil {
+		if etcdImage == "" {
+			return nil, nil, fmt.Errorf("either etcd resource or image vector should have %s image while deploying statefulset", common.Etcd)
```

Comment:
This will result in re-queueing of request and since the image is now nil this will keep getting re-queued right?

Reply:
Actually it'll only be re-queued if the images are nil and `etcd` or `etcd-backup-restore` images cannot be found in `charts/images.yaml`, see https://github.com/gardener/etcd-druid/blob/954f6ef747b7ffa4cc08d71f77e90aa4660b4677/controllers/etcd_controller.go#L992

Reply:
So if the images cannot be found, re-queue will also never help, so continuously re-queueing will block on worker all the time with no real advantage. Is there any alert raised?

Reply:
Yes, you are right. I added a TODO that this is supposed to be part of a validating webhook. No, there is no alert but the error message is put to `etcd.status.lastError`.